packages/app-runtime/src/modules/PushNotificationModule.ts

```diff
@@ -63,6 +64,41 @@ export class PushNotificationModule extends AppRuntimeModule<PushNotificationMod
         }
     }
 
+    private async syncEverythingUntilHasChanges(services: RuntimeServices): Promise<SyncEverythingResponse | undefined> {
+        const hasChanges = (value: SyncEverythingResponse): boolean =>
+            Object.values(value)
+                .filter((v) => Array.isArray(v))
+                .some((array) => array.length > 0);
+
+        const syncResult = await services.transportServices.account.syncEverything();
+
+        if (syncResult.isError) {
+            this.logger.error(syncResult.error);
+        } else if (hasChanges(syncResult.value)) {
+            return syncResult.value;
+        }
+
+        let retries = 0;
+        const maxRetries = 5;
```

Comment:
The first attempt already happened at this point.

```suggestion

        this.logger.info(`No changes found in sync attempt 1. Retrying...`);

        let retries = 1;
        const maxRetries = 5;
```

Reply:
Assuming there is no error, the first logging info will be `No changes found in sync attempt 1. Retrying...` even though there were already 2 sync attempts.

Reply:
updated - but in my way